src/main/java/io/github/perrymant/moneymaker/AccountRepository.java

```diff
@@ -0,0 +1,37 @@
+package io.github.perrymant.moneymaker;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.io.IOException;
+
+
+class AccountRepository {
+
+    Account read(String fileName) {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            return mapper.readValue(new File(fileName), Account.class);
+        } catch (IOException e) {
+            throw new FileIOException("FileIOException: couldn't read",e);
+        }
+    }
+
+    void write(Account data, String fileName) {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            mapper.writeValue(new File(fileName), data);
+        } catch (IOException e) {
+            throw new FileIOException("FileIOException: couldn't write",e);
+        }
+    }
+
+    private class FileIOException extends RuntimeException {
```

Comment:
Let's avoid private classes altogether. They're generally really awkward to deal with. This is especially the case with exceptions. A class that wants to catch this exception can't do so because it can't see this type.

 package private or public is best. Extract to own file too please.